docs/source/scheduler.md

```diff
@@ -34,15 +34,15 @@ If increasing core count doesn't decrease latency, that's a strong indicator tha
 
 _Multi-stream scheduling; requests execute in parallel and may utilize hardware resources better_
 
-Whereas the default scheduler will queue up requests made simultaneously and handle them serially, the multi-stream scheduler maintains a set of dropboxes where requests may be deposited and the requesting threads can wait. These dropboxes allow workers to find work from multiple sources when work from a single source would otherwise be scarce, maximizing throughput. When a request is complete, the requesting thread is awakened and returns the results to the caller.
+Whereas the default scheduler will queue up requests made simultaneously and handle them serially, the multi-stream scheduler allows multiple requests to be run in parallel. The num_streams argument to the Engine/Context classes controls how the multi-streams scheduler partitions up the machine. Each stream maps to a contiguous set of hardware threads. There is no sharing amongst the partitions and it is generally good practice make sure that the num_streams value evenly divides into your number of cores. By default num_streams is set to multiplex requests across L3 caches.
```

Comment:
I'd like us to clarify that the threads are numbered so that they are spatially compact in the sense that threads are closer if they share more of the memory hierarchy, but I don't have any better more precise suggestions without going way too in-depth